### ISSUE TITLE
Crash when login in with site address

### ISSUE THREAD

**Issue by JorgeMucientes** [woocommerce/woocommerce-android#8056]
**Describe the bug**

**Disclaimer**: this bug only affects users that have `FeatureFlag.REST_API` enabled in WooCommerce, which so far is only developers.

While reviewing [this hotfix](https://github.com/woocommerce/woocommerce-android/pull/8050), I noticed the users experience a similar issue when trying to log in using their site address if the site address contains suffixes like `wp-admin` or `wp-login.php`. When trying to match the entered site address with the `siteModelList` in this function from[ `SiteUtils` ](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/blob/9e07c99bb1142c607d99cbdd0fee665a127f9251/WordPressLoginFlow/src/main/java/org/wordpress/android/login/util/SiteUtils.java)

```kotlin 
    @Nullable
    private static SiteModel getSiteByMatchingUrl(List<SiteModel> siteModelList, String url) {
        if (siteModelList != null && !siteModelList.isEmpty()) {
            for (SiteModel siteModel : siteModelList) {
                String storedSiteUrl = UrlUtils.removeScheme(siteModel.getUrl()).replace("/", "");
                String incomingSiteUrl = UrlUtils.removeScheme(url).replace("/", "");
                if (storedSiteUrl.equalsIgnoreCase(incomingSiteUrl)) {
                    return siteModel;
                }
            }
        }
        return null;
    }
```
There won't be any match and thus null will be returned. If null is returned this recently added code in WooCommerce crashes: 

```kotlin
    override fun onSiteChanged(event: OnSiteChanged) {
        if (!event.isError && FeatureFlag.REST_API.isEnabled()) {
            val siteAddress = requireArguments().getString(ARG_INPUT_SITE_ADDRESS)
            val site = SiteUtils.getXMLRPCSiteByUrl(mSiteStore, siteAddress)!!
            selectedSite.set(site)
            mLoginListener.loggedInViaUsernamePassword(SiteUtils.getCurrentSiteIds(mSiteStore, false))
        } else {
            super.onSiteChanged(event)
        }
    }
```

Null pointer crash:
```
Caused by: java.lang.NullPointerException
	at com.woocommerce.android.ui.login.overrides.WooLoginUsernamePasswordFragment.onSiteChanged(WooLoginUsernamePasswordFragment.kt:49)
	at java.lang.reflect.Method.invoke(Native Method)
```

**Screenshots**

https://user-images.githubusercontent.com/2663464/209361919-96857d25-b67b-41af-9b79-3ef4e58d1c62.mp4

**Expected behaviour**

To fix the issue, we need to strip away any suffix from the site address, very similarly to what was done [here](https://github.com/woocommerce/woocommerce-android/pull/8050)

**Mobile Environment**
Please include:
- Device: Emulator
- Android version: 30
- WooCommerce Android version: 11.7

**Comment by hichamboushaba** [comment]
Good catch @JorgeMucientes and great investigation. This seems like an old bug and I wonder if it impacts WPAndroid as well 🤔.

As I want to focus on the REST API project, I won't have time to spend going deeper in the investigation, and I think we will make changes to this part soon to remove the XMLRPC checks. But if you have time to work on a fix, I'll be happy to review it, let's just keep Petros on the loop on this, as he is going to release a new RC build for 11.7 today.

**Comment by ParaskP7** [comment]
👋 @JorgeMucientes @hichamboushaba !

> But if you have time to work on a fix, I'll be happy to review it, let's just keep Petros on the loop on this, as he is going to release a new RC build for 11.7 today.

🙏 💯 🙇

**Comment by JorgeMucientes** [comment]
@hichamboushaba I can work on the fix this week. No prob. I might not be able to get it done today tough. 

> I wonder if it impacts WPAndroid as well 🤔.I'd say most likely,

 I'll check that as well. 


I think it'd be nice to have it for `11.7`, but I don't think is critical as it is a longstanding bug, so it shouldn't be a problem if it's released on `11.8`. Wdyt about the urgency of this login issue @pmusolino ?

cc: @ParaskP7 I think you can go head and generate new RC for `11.7` as I don't think I'll get this fix ready this morning.

**Comment by pmusolino** [comment]
@JorgeMucientes, given the other priorities I think it's ok to release it on 11.8 👍
